<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="description" default="ACC_ACCEL_FT_JONNY"/>
	
	<arg name="hwil" default="true"/>
	<arg name="target_speed" default="30.0"/>
	<arg name="max_headway" default="60.0"/>
	<arg name='readonly' default='true'/>

	<param name="description" value="$(arg description)"/>
		
	<param name="/max_headway" value="$(arg max_headway)"/>
	<param name="/target_speed" value="$(arg target_speed)"/>
	
	<param name="/readonly" value="$(arg readonly)"/>
	<param name="/hwil" value="$(arg hwil)"/>

	<node pkg="can_to_ros" type ="subs_fs" name="subs_fs" output="screen" if="$(arg hwil)"/> 
	<node pkg="can_to_ros" type="lead_info" name="lead_info" output="screen" if="$(arg hwil)"/>
	<node pkg="can_to_ros" type="simple_mini_car_from_lead_distance" name="simple_mini_car_from_lead_distance" output="screen" if="$(arg hwil)"/>

	<node pkg="acc_settings" type="acc_settings_node" name="acc_settings_node" output="screen">
    </node>	

	<node pkg="velocity_controller" type="velocity_controller_node" name="velocity_controller_node" output="screen" if="$(eval not readonly and hwil)">	
		<remap from="/vehicle/vel" to="/vel"/>	
	</node>
	
	<node pkg="velocity_controller" type="velocity_controller_node" name="velocity_controller_readonly_node" output="screen" if="$(eval readonly and hwil)">	
		<remap from="/vehicle/vel" to="/vel"/>	
		<remap from="/cmd_accel" to="/cmd_accel_null"/>
	</node>
	
	<node pkg="can_to_ros" type ="rosbag_record.sh" name="bashscript2" output="screen" args="$(arg description) $(arg hwil)" /> 
	</node>
	

</launch>
